src/invocation/ClusterService.ts

```diff
@@ -241,22 +256,22 @@ export class ClusterService implements Cluster {
         }
 
         for (const member of currentMembers) {
-            if (!deadMembers.delete(member.id())) {
+            if (this.memberListSnapshot.clusterUuid.equals(clusterUuid)) {
+                if (!deadMembers.delete(member.id())) {
+                    newMembers.push(member);
+                }
+            }
+            else {
                 newMembers.push(member);
             }
         }
-
+        
         const events = new Array<MembershipEvent>(deadMembers.size + newMembers.length);
         let index = 0;
 
         // removal events should be added before added events
         deadMembers.forEach((member) => {
             events[index++] = new MembershipEvent(member, MemberEvent.REMOVED, currentMembers);
-            const connection: Connection = connectionRegistry.getConnection(member.uuid);
-            if (connection != null) {
-                connection.close(null, new TargetDisconnectedError('The client has closed the connection to this '
-                    + 'member, after receiving a member left event from the cluster ' + connection));
-            }
```

Comment:
This logic is removed to the connection manager in the Java PR. Can you do the same? https://github.com/hazelcast/hazelcast/pull/20818/files?file-filters%5B%5D=.java&show-viewed-files=true&show-deleted-files=false#r841741898

Reply:
Fixed